openshift/dynamic/client.py

```diff
@@ -185,11 +185,12 @@ def patch(self, resource, body=None, name=None, namespace=None, **kwargs):
         if resource.namespaced:
             namespace = self.ensure_namespace(resource, namespace, body)
 
+        content_type = kwargs.pop('content_type',
+                                  self.client.select_header_content_type(['application/json-patch+json',
```

Comment:
I don't really understand why we're using select_header_content_type here, or why given that it will return `'application/json-patch+json'`,  it still does a strategic patch merge anyway but I'm trying to limit the scope of this change!

https://github.com/kubernetes-client/python/blob/master/kubernetes/client/api_client.py#L482-L485

Reply:
Yeah doesn't seem to make any sense. Only reason that it's using the `select_header_content_type` is that all the other uses of patch in the Kubernetes/OpenShift client did, but I never delved into what it was doing. Seems pretty useless here.

Reply:
@fabianvf I've simplified this further, seems reasonable to default to strategic.